src/pharmpy/model.py

```diff
@@ -109,9 +111,40 @@ def parameters(self):
 
     @parameters.setter
     def parameters(self, value):
+        if isinstance(value, Parameters):
+            inits = value.inits
+        else:
+            inits = value
+        if isinstance(inits, pd.Series):
+            inits = inits.to_dict()
+        if inits and not self.random_variables.validate_parameters(inits):
+            nearest = self.random_variables.nearest_valid_parameters(inits)
+            if nearest != inits:
+                before, after = self._compare_before_after_params(inits, nearest)
+                warnings.warn(
+                    f"Adjusting initial estimates to create positive semidefinite "
+                    f"omega/sigma matrices.\nBefore adjusting:  {before}.\n"
+                    f"After adjusting: {after}"
+                )
+                self._parameters.inits = nearest
+                return
```

Comment:
If value is Parameters cannot return here. Only inits have been updated

Reply:
Have changed to: 
```
if isinstance(value, Parameters):
   value.inits = nearest
else:
   value = nearest
```